base/linalg/cholesky.jl

```diff
@@ -460,6 +460,7 @@ function A_ldiv_B!(C::CholeskyPivoted, B::StridedMatrix)
     end
 end
 
+isposdef(C::CholeskyPivoted) = C.rank > 0
```

Comment:
As said on slack, this is not correct, as if `C.rank < n` then C is positive semidefinite or indefinite. Checking `info == 0` should be enough but a lapack lawyer would probably argue that according to the docs ("info = 0: algorithm completed successfully") this only guarantees `P'CP= U'U` for some U , so checking `C.info == 0 && diag(C[:U]) > 0` should give positive definiteness in the end.

Reply:
Thanks for pointing this out. I'll fix in a bit, I have ~493745 things happening today 😰 .

Reply:
Well, if it's sufficient, and you want to edit the PR while I'm dealing with my giant horrible to-do heap... 😉

Reply:
Either `rank == n` or `info == 0` should do the job here.

Reply:
That is what I also suspeckt. It is just annoying that the documentation does not provide this info in a clear way.